zerver/actions/typing.py

```diff
@@ -79,4 +79,20 @@ def do_send_stream_typing_notification(
 
     user_ids_to_notify = get_user_ids_for_streams({stream.id})[stream.id]
 
-    send_event(sender.realm, event, user_ids_to_notify)
+    try:
+        with open("/etc/zulip/stream_typing_notifications") as file:
+            portion_of_users_to_notify = float(file.read())  # nocoverage
+    except FileNotFoundError:
+        portion_of_users_to_notify = 0
+
+    new_size = int(len(user_ids_to_notify) * portion_of_users_to_notify)
+
+    count = 0
+    actual_user_ids_to_notify = set()
+    for user_id in user_ids_to_notify:
+        count += 1
+        if count > new_size:
+            break
+        actual_user_ids_to_notify.add(user_id)  # nocoverage
+
+    send_event(sender.realm, event, actual_user_ids_to_notify)
```

Comment:
This algorithm isn't random; it always sends to lower user IDs. I'm not at all convinced that we want roll a random number for each user ID, though -- the RNG queries might get expensive.

My thought had been to just check if `random.random() < portion_of_users_to_notify` and decide whether to send to 0 or all users based on that check.

Reply:
Done